web/src/stream_settings_containers.ts

```diff
@@ -0,0 +1,13 @@
+import $ from "jquery";
+import assert from "minimalistic-assert";
+
+import type {StreamSubscription} from "./sub_store";
+
+export function get_edit_container(sub: StreamSubscription | undefined): JQuery {
```

Comment:
In what situation will `sub` be `undefined`?

Reply:
That's what I'm not sure about. I checked every usage of `get_edit_container` and where their parameter `sub` come from. Theoratically if all the code works successfuly before reaching to `get_edit_container`, then `sub` should not be `undefined`. However, I'm not sure that "theoratically" part is correct, or is it too conservative to check `undefined` here?

Reply:
I think you are correct. `get_edit_container` itself always expects `sub` to be non-`undefined` because it will fail otherwise.